File: packages/sdk/examples/crewai-identity-gate.ts
import { AxiomSDK, createAxiomIDCrewAIToolDefinitions } from "../dist";

const sdk = new AxiomSDK({
  network: "mainnet",
  apiKey: process.env.AXIOMID_API_KEY,
});

const tools = createAxiomIDCrewAIToolDefinitions({
  sdk,
  agentDid: "did:axiom:crewai-research-agent",
  minimumTrustScore: 70,
});

async function authorizeCrewAITask() {
  const result = await tools.enforceSoulGate.run({
    did: "did:axiom:task-operator",
    purpose: "Authorize CrewAI research task",
  });

  if (!result.allowed) {
    throw new Error(result.reason);
  }

  return result.context;
}

authorizeCrewAITask()
  .then((context) => {
    console.log("CrewAI task authorized for", context.did);
  })
  .catch((error: unknown) => {
    console.error(error);
    process.exitCode = 1;
  });

File: packages/sdk/examples/crewai_identity_gate.py
"""CrewAI example that gates a task with AxiomID trust context."""

from crewai import Agent, Crew, Process, Task

from crewai_axiomid_tools import axiomid_enforce_soul_gate


identity_gatekeeper = Agent(
    role="AxiomID Identity Gatekeeper",
    goal="Verify that a DID has enough trust to run the assigned task.",
    backstory="You enforce AxiomID Soul Gate rules before other agents proceed.",
    tools=[axiomid_enforce_soul_gate],
    verbose=True,
)

gate_task = Task(
    description=(
        "Check whether did:axiom:operator can perform the research workflow. "
        "Use a minimum trust score of 70 and return the tool result."
    ),
    expected_output="A JSON gate result with allowed, reason, and trust context fields.",
    agent=identity_gatekeeper,
)

crew = Crew(
    agents=[identity_gatekeeper],
    tasks=[gate_task],
    process=Process.sequential,
)


if __name__ == "__main__":
    print(crew.kickoff())
